bump.lua

```diff
@@ -444,17 +474,21 @@ function World:addResponse(name, response)
   self.responses[name] = response
 end
 
-function World:project(item, x,y,w,h, goalX, goalY, filter)
+local EMPTY_TABLE = {}
+
+function World:project(item, x,y,w,h, goalX, goalY, filter, alreadyVisited)
   assertIsRect(x,y,w,h)
 
   goalX = goalX or x
   goalY = goalY or y
-  filter  = filter  or defaultFilter
+  filter = filter or defaultFilter
 
```

Comment:
We don't create the table for `collisions` until we've actually encountered a collision. This definitely cuts down on garbage collection, but also adds a branch to the logic. I believe it's worthwhile, but maybe you'll disagree.